src/rgw/rgw_op.h

```diff
@@ -1703,11 +1703,15 @@ static inline void rgw_get_request_metadata(CephContext *cct,
   }
 } /* rgw_get_request_metadata */
 
-static inline void encode_delete_at_attr(ceph::real_time delete_at,
+static inline void encode_delete_at_attr(boost::optional<ceph::real_time>& delete_at,
 					map<string, bufferlist>& attrs)
 {
-  if (real_clock::is_zero(delete_at)) {
-    return;
+  map<string, bufferlist>::iterator iter = attrs.find(RGW_ATTR_DELETE_AT);
+  if (delete_at == boost::none) {
```

Comment:
@rzarzynski ,I've updated them. I considered the http://tracker.ceph.com/issues/19074 ago, so I set `RGW_ATTR_DELETE_AT` to all objects before. But now there is another solution instead which is described below.